### ISSUE TITLE
support generic tests from dbt_utils when calculating coverage

### ISSUE THREAD

**Issue by aa-matthias** [slidoapp/dbt-coverage#32]
Our project is relying on tests defined in dbt_utils - is there a way of adding support for these as well?

https://github.com/dbt-labs/dbt-utils#unique_combination_of_columns-source

**Comment by mrshu** [comment]
@aa-matthias I presume these are currently not supported by `dbt-coverage` -- would that be the case?

Would you mind preparing a minimal example we could take a look at to better understand what would be needed to support these tests fully?

Thanks!

**Comment by aa-matthias** [comment]
Hi @mrshu - i've checked how tested columns are currently fetched in:

https://github.com/slidoapp/dbt-coverage/blob/main/dbt_coverage/__init__.py#L147

the problem is that for tests specified by dbt_utils kwargs are looking different - see the following example:

```json
            "test_metadata": {
                "name": "unique_combination_of_columns",
                "kwargs": {
                    "combination_of_columns": [
                        "id",
                        "version_id"
                    ],
                    "model": "{{ get_where_subquery(ref('sw6_base_order')) }}"
                },
                "namespace": "dbt_utils"
            },
```

**Comment by aa-matthias** [comment]
yes, i think this is the only one - we're currently making use of these as there are several models have no single column primary key / unique key. however it looks like it's best practice to add these so maybe we'll do that instead than also giving us enough coverage.

**Comment by sweco** [comment]
@aa-matthias, we are using the [official dbt-docs logic](https://github.com/dbt-labs/dbt-docs/blob/02731092389b18d69649fdc322d969b5d5b61b20/src/app/services/project_service.js#L155-L221) to parse the tests. If you generate the documentation, does the test get marked correctly in the UI for the columns?

<img width="74" alt="Screenshot 2022-11-21 at 17 37 22" src="https://user-images.githubusercontent.com/11132999/203110254-e76dfc39-1ac9-44b8-8d08-bf9ff6513992.png">
